### PR TITLE
WT-3133 Track operations by default in database root

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Comment by sulabhM** [comment]
autobuild: retest this please

**Comment by sulabhM** [comment]
autobuild: retest this please

**Comment by agorrod** [comment]
@sulabhM  I've pushed some changes here, but none of them fixed the Windows failure. I don't understand what's going on there.

I took your suggestion of sending all loging filename creation through a helper function after chasing a couple of mis-uses.

File: src/optrack/optrack.c [hunk 1]
```diff
@@ -53,19 +53,20 @@ int
...
 	conn = S2C(session);
 
 	if (!F_ISSET(conn, WT_CONN_OPTRACK))
 		return (WT_ERROR);
 
-	WT_RET(__wt_snprintf(optrack_fname,
-	    PATH_MAX, "%s/optrack.%" PRIuMAX ".%d",
-	    conn->optrack_path, conn->optrack_pid, session->id));
+	WT_RET(__wt_scr_alloc(session, 0, &buf));
+	WT_RET(__wt_filename_construct(session, conn->optrack_path,
```
Comment (agorrod) [inline src/optrack/optrack.c]: This should use WT_ERR and always free the buffer.

**Pull request merged**